specs/endpoint-security.spec.yml

```diff
@@ -45,6 +45,8 @@ inputs:
             - "--resources"
             - "endpoint-security-resources.zip"
           timeout: 600s
+          non_fatal_exit_codes:
+            - 28 # tamper protection enabled but could not (re)install
```

Comment:
This needs to be on the `check` operation as well or this will continue to fail in upgrades.

The `check` command does upgrades, the `install` command is responsible for installing for the first time.

Reply:
Did we confirm that this fix on its own is enough to mitigate the problem of exit code 28 on upgrade rollback by reproducing that known problem scenario?

Reply:
> This needs to be on the check operation as well or this will continue to fail in upgrades.

Okay, I will make a follow up PR to add this.

> Did we confirm that this fix on its own is enough to mitigate the problem of exit code 28 on upgrade rollback by reproducing that known problem scenario?

Yes, it does.  I just double checked it by building Agent off `main` (with `AGENT_PACKAGE_VERSION=9.1.0` so I could upgrade it) and following the test instructions from this PR.

Reply:
> > This needs to be on the check operation as well or this will continue to fail in upgrades.
> 
> Okay, I will make a follow up PR to add this.

https://github.com/elastic/elastic-agent/pull/9401